src/styles/withTheme.d.ts

```diff
@@ -4,7 +4,7 @@ export interface WithTheme {
   theme: Theme
 }
 
-declare const withTheme: <P = {}>() => (
+declare const withTheme: () => <P = {}>(
```

Comment:
While we're at it, I don't think this default is providing any value, might as well be just `<P>(`.